soccer/gameplay/tactics/advance_zone_midfielder.py

```diff
@@ -0,0 +1,138 @@
+import main
+import robocup
+import behavior
+import constants
+import enum
+import math
+
+import composite_behavior
+import skills.move
+import evaluation.ball
+import evaluation.passing_positioning
+import evaluation.field
+import evaluation.passing
+import evaluation.shooting
+import functools
+
+
+#2 offenders are related to do a certain event base off of great code
+class AdvanceZoneMidfielder(composite_behavior.CompositeBehavior):
+    # Weights for the general field positioning
+    FIELD_POS_WEIGHTS = (0.01, 3, 0.02)
+    # Weights for finding best pass
+    PASSING_WEIGHTS = (2, 2, 15, 1)
+    # Initial arguements for the nelder mead optimization in passing positioning
+    NELDER_MEAD_ARGS = (robocup.Point(0.5, 2), \
+                        robocup.Point(0.01, 0.01), 1, 2, \
+                        0.75, 0.5, 50, 1, 0.1)
```

Comment:
We might want to pass these in from `adaptive_formation` too since these constants significantly change the behavior of  the passing positioning method.